### PR TITLE
[#62] Lock the AWS provider version and Terraform version

### PR DESCRIPTION
**Pull request by hoangmirs**
Resolves https://github.com/nimblehq/infrastructure-templates/issues/62

## What happened 👀

- Add versions for terraform and providers
- Move `terraform` and `provider` blocks into `providers.tf` file

## Insight 📝

N/A

## Proof Of Work 📹

N/A

### CODE REVIEW

File: skeleton/aws/main.tf [hunk 1]
```diff
@@ -1,24 +1,3 @@
-terraform {
-  cloud {
-    organization = "organization"
-
-    workspaces {
-      name = "terraform_workspace"
-    }
-  }
-}
```
Comment (malparty) [inline skeleton/aws/main.tf:9]: This part seems to belong to the main, more than the provider part, no? If not, can you share more insights about what makes it belong to "providers" (I'm still discovering Terraform 😇 )
Thanks! :)

**Review by malparty — CHANGES_REQUESTED** [review]
A small question to ensure I understand it well :)

File: skeleton/aws/main.tf (hunk 1 above)
Comment (hoangmirs) [reply skeleton/aws/main.tf:9]: We don't have any convention for this, but as I just want to move the provider into a separate file, this block also has `required_providers` attribute, so I moved it as well.
Then later, when adding a new provider, we just need to modify that file only. What do you think?
[The standard files](https://www.terraform.io/language/modules/develop/structure) don't have `providers.tf` file :D

Comment (malparty) [reply skeleton/aws/main.tf:9]: It seems like there can be multiple `terraform` blocks... if yes, should we make it like this? (to be tested :P ) 
 
 
image

Comment (malparty) [reply skeleton/aws/main.tf:9]: If not, let's go for the "all in providers" solution as you initially implemented ^^,

Comment (hoangmirs) [reply skeleton/aws/main.tf]: Good catch, we can use multiple terraform blocks 🙌 
I fixed in https://github.com/nimblehq/infrastructure-templates/pull/69/commits/61af80a5a8843e83abf9bd6d0c3c938fb3a3971d

**Review by andyduong1920 — APPROVED** [review]

**Pull request merged**